ground/src/main/java/com/google/android/ground/repository/TermsOfServiceRepository.kt

```diff
@@ -37,12 +37,13 @@ constructor(
   var isTermsOfServiceAccepted: Boolean by localValueStore::isTermsOfServiceAccepted
 
   suspend fun getTermsOfService(): TermsOfService? =
-    runCatching {
-        // TODO(#1691): Maybe parse the exception and display to the user.
-        remoteDataStore
-          .loadTermsOfService()
-          .timeout(LOAD_REMOTE_SURVEY_TERMS_OF_SERVICE_TIMEOUT_SECS, TimeUnit.SECONDS)
-          .awaitSingleOrNull()
+    // TODO(#1691): Maybe parse the exception and display to the user.
+    withTimeoutOrNull(LOAD_REMOTE_SURVEY_TERMS_OF_SERVICE_TIMEOUT_MILLIS) {
+      try {
```

Comment:
Can we handle this downstream? See comment https://github.com/google/ground-android/issues/1691#issuecomment-1686372000

Reply:
Done